src/main/java/org/broadinstitute/hellbender/tools/recalibration/RecalUtils.java

```diff
@@ -278,27 +163,12 @@ protected static CsvPrinter csvPrinter(final File out, final Covariate ... c)
     public static void generateCsv(final File out, final Map<String, RecalibrationReport> reports)
             throws FileNotFoundException {
         if (reports.size() == 0) {
-            writeCsv(out, reports, new Covariate[0]);
+            throw new GATKException("no reports");
         } else {
             final Iterator<RecalibrationReport> rit = reports.values().iterator();
             final RecalibrationReport first = rit.next();
-            final Covariate[] firstCovariates = first.getRequestedCovariates();
-            final Set<Covariate> covariates = new LinkedHashSet<>();
-            Utils.addAll(covariates,firstCovariates);
-            while (rit.hasNext() && covariates.size() > 0) {
-                final Covariate[] nextCovariates = rit.next().getRequestedCovariates();
-                final Set<String> nextCovariateNames = new LinkedHashSet<String>(nextCovariates.length);
-                for (final Covariate nc : nextCovariates) {
-                    nextCovariateNames.add(nc.getClass().getSimpleName());
-                }
-                final Iterator<Covariate> cit = covariates.iterator();
-                while (cit.hasNext()) {
-                    if (!nextCovariateNames.contains(cit.next().getClass().getSimpleName())) {
-                        cit.remove();
-                    }
-                }
-            }
-            writeCsv(out, reports, covariates.toArray(new Covariate[covariates.size()]));
+            final StandardCovariateList covariates = first.getCovariates();
```

Comment:
Do you even need to get this out of `first`? Eg., could you just create a new `StandardCovariateList` here (or use a singleton instance)? Getting it out of `first` implies that the covariates used could depend on the input `RecalibrationReports`, which is no longer the case.